packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -89,6 +89,12 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
     assert(!attached);
   }
 
+  /// Find the last [AnnotatedRegionLayer] which contains [regionOffset] with a
+  /// value of Type [type].
+  ///
+  /// Returns null if no matching region is found.
+  Object findRegion(Offset regionOffset, Type type);
```

Comment:
For type safety, you can change the signature of this method to:

```dart
   T findRegion<T>(Offset regionOffset) ;
```

Reply:
"findRegion" sounds like it returns a region... maybe "findValue"?
If you made the argument named "at" you could have the call sites look like:

```dart
  MyData data = layer.findValue<MyData>(at: const Offset(20.0, 40.0));
```

Reply:
Or even maybe just "find", as in:

```dart
  MyData data = layer.find<MyData>(at: const Offset(20.0, 40.0));
```